fedimint-client/src/db.rs

```diff
@@ -35,7 +35,7 @@ pub enum DbKeyPrefix {
     ChronologicalOperationLog = 0x2d,
     CommonApiVersionCache = 0x2e,
     ClientConfig = 0x2f,
-    ClientInviteCode = 0x30,
+    ClientInviteCode = 0x30, // Unused; clean out remnant data before re-using!
```

Comment:
We don't need a migration. We can just always delete this key on start, and over time it will get deleted from all clients and rotated out of existance.

Reply:
I considered this, but we could never be sure could we?

Reply:
> We don't need a migration. We can just always delete this key on module start, and over time it will get deleted from all clients and rotated out of existence.

This PR has an example of how to do this https://github.com/fedimint/fedimint/pull/4427/files#diff-d754f9b1f77285e47d11bdecd4e3df652f846e5cedd12ded095ae84816a64cc2

Reply:
> I considered this, but we could never be sure could we?

I mean - we would still keep `ClientInviteCode` variant in the code with annotation not to use it, but instead of not to the future self: "clean out remnant data before re-using!" we would just do it eagerly.


After long enough time for all practical purposes we'd be sure no supported client can possiblity still have it there.

I doubt it is going to matter in practice, but conceptually it's strictly better approach.

Reply:
`#[deprecated]` ?

Reply:
> I doubt it is going to matter in practice, but conceptually it's strictly better approach.

I agree with both, would prefer to get this merged now though.

Reply:
Deleting unconditionally is a hack imo and DB migrations are basically free to add.

If we eventually remove old migrations the client can refuse to start if a too old DB is encountered, but that only works if all DB migrations are modeled properly.